src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java

```diff
@@ -425,13 +432,26 @@ public void valueChanged(ListSelectionEvent e) {
 				}});
 			
 			uninstalledAddOnsTable.setModel(uninstalledAddOnsModel);
-			uninstalledAddOnsTable.getColumnModel().getColumn(0).setMaxWidth(20);
-			uninstalledAddOnsTable.getColumnModel().getColumn(1).setPreferredWidth(50);
-			uninstalledAddOnsTable.getColumnModel().getColumn(2).setPreferredWidth(150);
-			uninstalledAddOnsTable.getColumnModel().getColumn(3).setPreferredWidth(300);
-			uninstalledAddOnsTable.getColumnModel().getColumn(4).setPreferredWidth(60);
-			uninstalledAddOnsTable.getColumnModel().getColumn(5).setPreferredWidth(40);
-
+			uninstalledAddOnsTable.getColumnModel().getColumn(0).setMaxWidth(20);//Icon
+			uninstalledAddOnsTable.getColumnExt(0).setSortable(false); //Icon doesn't need sorting
+			uninstalledAddOnsTable.getColumnModel().getColumn(1).setPreferredWidth(50);//Status
+			uninstalledAddOnsTable.getColumnModel().getColumn(2).setPreferredWidth(150);//Name
+			uninstalledAddOnsTable.getColumnModel().getColumn(3).setPreferredWidth(300);//Description
+			uninstalledAddOnsTable.getColumnExt(3).setSortable(false);//Description doesn't need sorting
+			uninstalledAddOnsTable.getColumnModel().getColumn(4).setPreferredWidth(60);//Update (version number)
+			uninstalledAddOnsTable.getColumnExt(4).setSortable(false);//Update doesn't need sorting
+			uninstalledAddOnsTable.getColumnModel().getColumn(5).setPreferredWidth(40);//Checkbox
+			uninstalledAddOnsTable.getColumnExt(5).setSortable(false);//Checkbox doesn't need sorting
+
+
+			uninstalledAddOnsTable.setAutoCreateRowSorter(true);
+           
+            //Default sort by status (column 1) descending (Release, Beta, Alpha), and name (column 2) ascending 
+            List<RowSorter.SortKey> sortKeys = new ArrayList<RowSorter.SortKey>(2);
+            sortKeys.add(new RowSorter.SortKey(1, SortOrder.DESCENDING));

```

Comment:
Ok I'm finally back to this. I'm left with a ton of questions:

1.  You mentioned dropping the comparator in the model but it seems to be required by a  call to the super constructor at [L62](https://github.com/zaproxy/zaproxy/blob/develop/src/org/zaproxy/zap/extension/autoupdate/UninstalledAddOnsTableModel.java#L62). Would we drop some part of StatusUI in this place or is there a way to avoid the requirement?
2. Where do you see StatusUI fitting in? I'm thinking either: org.zaproxy.zap.utils or org.zaproxy.zap.view.
3. You mention caching the StatusUI info. What do you mean by "caching".
4. You mention returning as part of getValueAt(int, int). Were you intending that info to replace the [current cfu.status and wrapper/status info](https://github.com/zaproxy/zaproxy/blob/develop/src/org/zaproxy/zap/extension/autoupdate/UninstalledAddOnsTableModel.java#L98)?
5. In the implementation you handed me....thanks by the way! :thumbsup: Where does stringRepresentation ever get set? Would it be passed in to the constructor based on a reference to an addon or addon wrapper?